### DISCUSSION TITLE
hooks.js not getting generated?

### DISCUSSION THREAD

**Discussion by davinstewart** [sveltejs/kit#12917] [Q&A] [answered]
Beginner question here ... src/hooks.js is not getting auto-generated when I create a new sveltekit minimal project.  Is this normal?  From reading all of the documentation, it seems like it should be getting generated.  No errors are thrown during project creation so I'm very confused.  I've searched everywhere I can think of for an explanation and found nothing.

Please help!

**Comment by brunnerh** [comment]
> Is this normal?

Yes.

There are many optional files that are not generated & hooks are an advanced thing that quite a few applications may not even need (they are also described in the "advanced" section of the docs).

**Reply by davinstewart** [reply]
Thank you for the quick response.  So if I wanted to start using hooks.js then I'd just create the file and it'll automatically get picked up by the platform or is there more configuration that I'd need to do?

**Reply by brunnerh** [reply, marked as answer]
You just need to create the file in the right place.

**Comment by davinstewart** [comment]
Excellent!  Thank you and much appreciated!
